### PR TITLE
:bug: Make Yuyuko unable to do SoulDrain DrawCard when SP Flandre los…

### PR DESCRIPTION
**Pull request by CoffeeYin-CY**
An old story for SP Flandre, same as that dispute between SP Flan and Shizuha:

emit_event: action_shootdown DecayAction
applying action DecayAction, current hybrid_stack: [THBattleKOFBootstrap, <thb.characters.sp_flandre.DestructionImpulseHandler object at 0x0A80A890>, DestructionImpulseAction, <thb.characters.shizuha.DecayDamageHandler object at 0x0A80A750>]
emit_event: action_apply DecayAction

As a result, if PlayerTurn isn't in action-stack, do not do Decay.
Also, SP's HP goes down to 0, yuyuko draws card: this shall get mended in yuyuko.py, which makes this handler deactivate when now is not in any character's turn.

So, both append one line:
```
if not any(isinstance(a, PlayerTurn) for a in g.action_stack): return act
```

Which passes local server test.

### CODE REVIEW

**Comment by feisuzhu** [comment]
改 SpFlandre 吧，听的事件不对，改听 `action_apply FinalizeStage` 吧

**Comment by CoffeeYin-CY** [comment]
其实我觉得给SP芙兰这样写也没有问题……
况且如果不这样改，后面sp与uuz，sp与debut又要经历重测……
而且又要改秋静叶描述“当前回合角色加上‘凋零’标记”这类……

**Comment by feisuzhu** [comment]
最早是没有 PrepareStage 和 FinalizeStage 的，因为用这种方式太多导致的问题太多才改成现在的样子。这个应该是历史遗留。

重测应该是不用了，这样修改是比原来更正规一点，现在也没有阻止`FinalizeStage`的逻辑。

**Comment by CoffeeYin-CY** [comment]
感觉这个是可以的，只是改了uuz的描述和对于是否在PlayerTurn里的检查。SP芙兰的时机会另改，这个似乎可以合并……

File: src/thb/characters/yuyuko.py [hunk 1]
```diff
@@ -124,6 +124,7 @@ def handle(self, evt_type, act):
             else:
                 return act
 
+            if not any(isinstance(a, PlayerTurn) for a in g.action_stack): return act
             for a in reversed(g.action_stack):
```
Comment (feisuzhu) [inline src/thb/characters/yuyuko.py]:
```python
for a in reversed(...):
    ...
else:
   return act
```

Comment (feisuzhu) [reply src/thb/characters/yuyuko.py]: 原来的地方还要加上 `else break`

**Pull request merged**